pkg/streamingpromql/limiting/memory_consumption_test.go

```diff
@@ -117,3 +119,81 @@ func createRejectedMetric() (*prometheus.Registry, prometheus.Counter) {
 
 	return reg, metric
 }
+
+func BenchmarkMemoryConsumptionTracker(b *testing.B) {
+	b.Run("no limits single threaded", func(b *testing.B) {
+		l := NewMemoryConsumptionTracker(0, nil)
+		for i := 0; i < b.N; i++ {
```

Comment:
[nit] Could use the new `b.Loop()` method here and below

Reply:
We still declare `go 1.23.7` in `go.mod` so the linter warns about using `b.Loop()`. I've added timer resets to the loops.